src/drift-monitor.ts

```diff
@@ -49,41 +49,37 @@ export class DriftMonitor extends Construct {
 
   public readonly alarm: Alarm;
 
-  constructor(scope: Construct, id: string, props: DriftMonitorProps) {
+  constructor(scope: Construct, id: string, props?: DriftMonitorProps) {
```

Comment:
Then you don't have to use `?` everywhere...

```suggestion
  constructor(scope: Construct, id: string, props: DriftMonitorProps = {}) {
```